### PR TITLE
Thang bottom bar

### PR DESCRIPTION
**Pull request by huynhthanhthang1296**

### CODE REVIEW

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    public ViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        if (position == 0){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            return new TabHomeFragment();
+        } else if (position == 1){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            return new TabSearchFragment();
+        } else if (position == 2){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            return new TabFavoriteFragment();
+        } else if (position == 3){
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoritePresenter.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Listens to user actions from the UI ({@link TabFavoriteFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabFavoritePresenter implements TabFavoriteContract.Presenter {
+    private static final String TAG = TabFavoritePresenter.class.getName();
+
+    private final TabFavoriteContract.View mView;
+
+    public TabFavoritePresenter(TabFavoriteContract.View view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoritePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/TabHomePresenter.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Listens to user actions from the UI ({@link TabHomeFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabHomePresenter implements TabHomeContract.Presenter {
+    private static final String TAG = TabHomePresenter.class.getName();
+
+    private final TabHomeContract.View mView;
+
+    public TabHomePresenter(TabHomeContract.View view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/TabHomePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfilePresenter.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Listens to user actions from the UI ({@link TabProfileFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabProfilePresenter implements TabProfileContract.Presenter {
+    private static final String TAG = TabProfilePresenter.class.getName();
+
+    private final TabProfileContract.View mView;
+
+    public TabProfilePresenter(TabProfileContract.View view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfilePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Listens to user actions from the UI ({@link TabSearchFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabSearchPresenter implements TabSearchContract.Presenter {
+    private static final String TAG = TabSearchPresenter.class.getName();
+
+    private final TabSearchContract.View mView;
+
+    public TabSearchPresenter(TabSearchContract.View view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoriteContract.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.screen.BaseView;
+
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface TabFavoriteContract {
+    /**
+     * View.
+     */
+    interface View extends BaseView {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoriteContract.java]: WARNING: [ShortClassName] Avoid short class names like View
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/TabHomeContract.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.screen.BaseView;
+
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface TabHomeContract {
+    /**
+     * View.
+     */
+    interface View extends BaseView {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/TabHomeContract.java]: WARNING: [ShortClassName] Avoid short class names like View
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfileContract.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.screen.BaseView;
+
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface TabProfileContract {
+    /**
+     * View.
+     */
+    interface View extends BaseView {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfileContract.java]: WARNING: [ShortClassName] Avoid short class names like View
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchContract.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+import com.fstyle.androidtrainning.screen.BaseView;
+
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface TabSearchContract {
+    /**
+     * View.
+     */
+    interface View extends BaseView {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchContract.java]: WARNING: [ShortClassName] Avoid short class names like View
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/fstyle/androidtrainning/UnSwipeViewPager.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        super(context, attrs);
+    }
+
+    @Override
+    public boolean onInterceptTouchEvent(MotionEvent ev) {
+        return false;
+    }
+
+    @Override
+    public boolean onTouchEvent(MotionEvent ev) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/UnSwipeViewPager.java:35]: WARNING: [ClickableViewAccessibility] Custom view `UnSwipeViewPager` overrides `onTouchEvent` but not `performClick`
 If a `View` that overrides `onTouchEvent` or uses an `OnTouchListener` does not also implement `performClick` and call it when clicks are detected, the `View` may not handle accessibility actions properly. Logic handling the click actions should ideally be placed in `View#performClick` as some accessibility services invoke `performClick` when a click action should occur.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoritePresenter.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
+package com.fstyle.androidtrainning.screen.tabfavorite;
+
+/**
+ * Listens to user actions from the UI ({@link TabFavoriteFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabFavoritePresenter implements TabFavoriteContract.Presenter {
+    private static final String TAG = TabFavoritePresenter.class.getName();
+
+    public final TabFavoriteContract.FavoriteView mView;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoritePresenter.java]: WARNING: Variable 'mView' must be private and have accessor methods.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoritePresenter.java [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+    public TabFavoritePresenter(TabFavoriteContract.FavoriteView view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabfavorite/TabFavoritePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/TabHomePresenter.java [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Listens to user actions from the UI ({@link TabHomeFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabHomePresenter implements TabHomeContract.Presenter {
+    public static final String TAG = TabHomePresenter.class.getName();
+
+    private final TabHomeContract.HomeView mView;
+
+    public TabHomePresenter(TabHomeContract.HomeView view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/TabHomePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfilePresenter.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
+package com.fstyle.androidtrainning.screen.tabprofile;
+
+/**
+ * Listens to user actions from the UI ({@link TabProfileFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabProfilePresenter implements TabProfileContract.Presenter {
+    private static final String TAG = TabProfilePresenter.class.getName();
+
+    public final TabProfileContract.ProfileView mView;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfilePresenter.java]: WARNING: Variable 'mView' must be private and have accessor methods.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfilePresenter.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+    public TabProfilePresenter(TabProfileContract.ProfileView view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabprofile/TabProfilePresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java [hunk 19]
```diff
@@ -0,0 +1,23 @@
+package com.fstyle.androidtrainning.screen.tabsearch;
+
+/**
+ * Listens to user actions from the UI ({@link TabSearchFragment}), retrieves the data and updates
+ * the UI as required.
+ */
+final class TabSearchPresenter implements TabSearchContract.Presenter {
+    private static final String TAG = TabSearchPresenter.class.getName();
+
+    public final TabSearchContract.SearchView mView;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java]: WARNING: Variable 'mView' must be private and have accessor methods.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+    public TabSearchPresenter(TabSearchContract.SearchView view) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java]: ERROR: Redundant 'public' modifier.

File: app/src/main/java/com/fstyle/androidtrainning/UnSwipeViewPager.java [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+        return false;
+    }
+
+    @Override
+    public boolean canScrollHorizontally(int direction) {
+        return false;
+    }
+
+    @Override
+    public boolean onTouchEvent(MotionEvent ev) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/UnSwipeViewPager.java:35]: WARNING: [ClickableViewAccessibility] `UnSwipeViewPager#onTouchEvent` should call `UnSwipeViewPager#performClick` when a click is detected
 If a `View` that overrides `onTouchEvent` or uses an `OnTouchListener` does not also implement `performClick` and call it when clicks are detected, the `View` may not handle accessibility actions properly. Logic handling the click actions should ideally be placed in `View#performClick` as some accessibility services invoke `performClick` when a click action should occur.

File: app/src/main/res/values/dimens.xml [hunk 22]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="dp_55">55dp</dimen>
+</resources>
```
Comment (tranminhtri911) [inline app/src/main/res/values/dimens.xml]: add end of line

File: app/src/main/res/layout/fragment_tabsearch.xml [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.fstyle.structure_android.screen.main.MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Search"
```
Comment (tranminhtri911) [inline app/src/main/res/layout/fragment_tabsearch.xml]: hard code

File: app/src/main/res/layout/fragment_tabprofile.xml [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.fstyle.structure_android.screen.main.MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Profile"
```
Comment (tranminhtri911) [inline app/src/main/res/layout/fragment_tabprofile.xml]: hard code

File: app/src/main/res/layout/fragment_tabhome.xml [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.fstyle.structure_android.screen.main.MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Home"
```
Comment (tranminhtri911) [inline app/src/main/res/layout/fragment_tabhome.xml]: hard code

File: app/src/main/res/layout/fragment_tabfavorite.xml [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.fstyle.structure_android.screen.main.MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Favorite"
```
Comment (tranminhtri911) [inline app/src/main/res/layout/fragment_tabfavorite.xml]: hard code

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    public ViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        if (position == 0) {
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: Define variable 0,1,2,3
use swith case

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+            return new TabFavoriteFragment();
+        } else if (position == 3) {
+            return new TabProfileFragment();
+        }
+        return null;
+    }
+
+    @Override
+    public int getCount() {
+        return 4;
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: define total_tab

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 29]
```diff
@@ -1,13 +1,53 @@
...
+
+    private void handleEvents() {
+        mBottombar.setOnTabSelectListener(this);
+    }
+
+    @Override
+    public void onTabSelected(int tabId) {
+        switch (tabId){
+            case R.id.tabhome:
+                mViewPager.setCurrentItem(0);
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: please define tab 0,1 ,2 with IntDef same above :
@intdef({
Tab.HOME, Tab.ABC, Tab.XYZ
})
public @interface Tab{
int HOME= 10;
int ABC= 11;
int XYZ = 12;
}

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 30]
```diff
@@ -1,13 +1,53 @@
...
+            case R.id.tabsearch:
+                mViewPager.setCurrentItem(1);
+                break;
+            case R.id.tabfavorite:
+                mViewPager.setCurrentItem(2);
+                break;
+            case R.id.tabprofile:
+                mViewPager.setCurrentItem(3);
+                break;
+
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: default

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 31]
```diff
@@ -0,0 +1,43 @@
...
+import com.fstyle.androidtrainning.screen.tabhome.TabHomeFragment;
+import com.fstyle.androidtrainning.screen.tabprofile.TabProfileFragment;
+import com.fstyle.androidtrainning.screen.tabsearch.TabSearchFragment;
+
+/**
+ * Created by huynh on 12/12/2017.
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private static int TOTAL_TAB = 4;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: WARNING: Name 'TOTAL_TAB' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 32]
```diff
@@ -1,13 +1,72 @@
...
+                break;
+            case R.id.tabprofile:
+                mViewPager.setCurrentItem(PROFILE);
+                break;
+            default:
+        }
+    }
+
+
+    @IntDef({HOME, Tab.SEARCH, Tab.FAVORITE, Tab.PROFILE})
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.FAVORITE' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.FAVORITE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.PROFILE' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.PROFILE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.SEARCH' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.SEARCH'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java (hunk 31 above)
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+import com.fstyle.androidtrainning.screen.tabhome.TabHomeFragment;
+import com.fstyle.androidtrainning.screen.tabprofile.TabProfileFragment;
+import com.fstyle.androidtrainning.screen.tabsearch.TabSearchFragment;
+
+/**
+ * Created by huynh on 12/12/2017.
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private static int total_tab = 4;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: WARNING: Name 'total_tab' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 34]
```diff
@@ -1,13 +1,72 @@
...
 
-public class MainActivity extends AppCompatActivity {
+import com.fstyle.androidtrainning.screen.ViewPagerAdapter;
+import com.roughike.bottombar.BottomBar;
+import com.roughike.bottombar.OnTabSelectListener;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import static com.fstyle.androidtrainning.MainActivity.Tab.FAVORITE;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.fstyle.androidtrainning.MainActivity.Tab.FAVORITE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 35]
```diff
@@ -1,13 +1,72 @@
...
+import static com.fstyle.androidtrainning.MainActivity.Tab.HOME;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.fstyle.androidtrainning.MainActivity.Tab.HOME'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 36]
```diff
@@ -1,13 +1,72 @@
...
+import static com.fstyle.androidtrainning.MainActivity.Tab.PROFILE;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.fstyle.androidtrainning.MainActivity.Tab.PROFILE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 37]
```diff
@@ -1,13 +1,72 @@
...
+import static com.fstyle.androidtrainning.MainActivity.Tab.SEARCH;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.fstyle.androidtrainning.MainActivity.Tab.SEARCH'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 38]
```diff
@@ -1,13 +1,72 @@
...
+
+    private void handleEvents() {
+        mBottombar.setOnTabSelectListener(this);
+    }
+
+    @Override
+    public void onTabSelected(int tabId) {
+        switch (tabId) {
+            case R.id.tabhome:
+                mViewPager.setCurrentItem(Tab.HOME);
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java:36]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.HOME' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.HOME'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 39]
```diff
@@ -1,13 +1,72 @@
...
+                break;
+            case R.id.tabsearch:
+                mViewPager.setCurrentItem(Tab.SEARCH);
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java:39]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.SEARCH' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.SEARCH'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 40]
```diff
@@ -1,13 +1,72 @@
...
+                break;
+            case R.id.tabfavorite:
+                mViewPager.setCurrentItem(Tab.FAVORITE);
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java:42]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.FAVORITE' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.FAVORITE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 41]
```diff
@@ -1,13 +1,72 @@
...
+                break;
+            case R.id.tabprofile:
+                mViewPager.setCurrentItem(Tab.PROFILE);
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java:45]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.PROFILE' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.PROFILE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 42]
```diff
@@ -1,13 +1,72 @@
...
+                break;
+            default:
+        }
+    }
+
+
+    @IntDef({Tab.HOME, Tab.SEARCH, Tab.FAVORITE, Tab.PROFILE})
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.FAVORITE' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.FAVORITE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.HOME' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.HOME'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.PROFILE' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.PROFILE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Tab.SEARCH' due to existing static import 'com.fstyle.androidtrainning.MainActivity.Tab.SEARCH'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java (hunk 33 above)
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: WARNING: Name 'total_tab' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java [hunk 43]
```diff
@@ -0,0 +1,43 @@
...
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private static final int TOTAL_TAB = 4;
+    public ViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        switch (position){
+            case 0:
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: please define 0,1,2 same class MainActivity

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 44]
```diff
@@ -1,13 +1,65 @@
...
+    private ViewPager mViewPager;
+    private BottomBar mBottombar;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        initViews();
+        final ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager());
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: remove down method initView

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 45]
```diff
@@ -1,13 +1,65 @@
...
+        mViewPager.setAdapter(viewPagerAdapter);
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java:23]: above

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 46]
```diff
@@ -1,13 +1,65 @@
...
+        handleEvents();
+    }
+
+    private void initViews() {
+        mViewPager = findViewById(R.id.unSwipViewPager);
+        mBottombar = findViewById(R.id.bottomBar);
+    }
+
+    private void handleEvents() {
+        mBottombar.setOnTabSelectListener(this);
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: above

File: app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java (hunk 43 above)
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/screen/ViewPagerAdapter.java]: replace 0 = MainActivity.Tab.Home

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java (hunk 46 above)
Comment (tranminhtri911) [reply app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: cho vào method initView luôn đi em

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 47]
```diff
@@ -1,13 +1,61 @@
...
+public class MainActivity extends AppCompatActivity implements OnTabSelectListener {
+    private ViewPager mViewPager;
+    private BottomBar mBottombar;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        final ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager());
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java]: remove xuống initView toàn bộ chỗ này đi em

File: app/src/main/java/com/fstyle/androidtrainning/MainActivity.java [hunk 48]
```diff
@@ -1,13 +1,61 @@
...
+            case R.id.tabsearch:
+                mViewPager.setCurrentItem(Tab.SEARCH);
+                break;
+            case R.id.tabfavorite:
+                mViewPager.setCurrentItem(Tab.FAVORITE);
+                break;
+            case R.id.tabprofile:
+                mViewPager.setCurrentItem(Tab.PROFILE);
+                break;
+            default:
```
Comment (tranminhtri911) [inline app/src/main/java/com/fstyle/androidtrainning/MainActivity.java:47]: add break nữa

**Comment by tranminhtri911** [comment]
Ok

**Pull request merged**